### PR TITLE
vo_gpu: use interpolation-threshold on integer multiples

### PR DESCRIPTION
**Pull request by Dudemanguy**
Currently, the --interpolation-threshold option only ever applies if the
vsync ratio is close to 1 (i.e. 60fps video on a 60hz display). This
changes the logic of interpolation-threshold to instead apply to the
nearest integer of the vsync ratio (i.e. 30fps video on a 60hz display
would disable the use of interpolation by default). -1 still disables
this threshold and always enables interpolation like before.

Alternative approach to #8934.

Edit: was suggested in IRC to use `abs(ratio - round(ratio) / ratio` instead. I'll update this later.

### CODE REVIEW

**Comment by haasn** [comment]
Some thoughts: 

* I personally *do* want interpolation for 30 fps content on a 60 fps monitor, but not 60 fps content on a 60 fps monitor. The logic here makes sense for `oversample` interpolation, but no other modes, IMO.
* `--vo=gpu-next` uses the same logic as current `--vo=gpu` so you might want to, at the very least, document that there will be a discrepancy.
* If I add this to `--vo=gpu-next` it would almost surely be as a separate `bool`.

In conclusion I think my advice would be to either make this a separate toggle, or better yet, make it depend on the interpolation mode requested. (Enabled for `oversample`, disabled for the others).

**Comment by Dudemanguy** [comment]
This kind of went stale and I'm not really sure about this so I'll go ahead and close this.
